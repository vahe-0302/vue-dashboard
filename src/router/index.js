import Vue from 'vue'
import Router from 'vue-router'
import Head from 'vue-head'
import { isNil } from 'lodash'
import Home from '@/views/Home'
import CheckLogin from '@/views/CheckLogin'
import store from '@/store'

Vue.use(Router)

/* If you don't know about VueHead, please refer to https://github.com/ktquez/vue-head */

Vue.use(Head, {
  complement: process.env.VUE_APP_TITLE
})

/* If you don't know about VueRouter, please refer to https://router.vuejs.org/ */

const router = new Router({
  mode: 'history',
  base: process.env.BASE_URL,
  routes: [
    {
      path: '/',
      name: 'Home',
      component: Home,
      meta: {
        authNotRequired: true
      }
    },
    {
      path: '/index.html',
      redirect: { name: 'Home' }
    },
    {
      path: '/check-login',
      name: 'check-login',
      component: CheckLogin,
      meta: {
        authNotRequired: true
      }
    },
    {
      path: '/login',
      name: 'login',
      component: () =>
        import(/* webpackChunkName: "client-chunk-login" */ '@/views/Login.vue'),
      meta: {
        authNotRequired: true
      }
    },
    {
      path: '/feedstocks',
      name: 'Feedstocks',
      component: () =>
        import(/* webpackChunkName: "client-chunk-feedstocks-categories" */ '@/views/FeedstocksCategories.vue'),
      props: true,
      meta: {
        authNotRequired: true
      }
    },
    {
      path: '/feedstocks/:categoryId',
      name: 'Material',
      component: () =>
        import(/* webpackChunkName: "client-chunk-feedstocks" */ '@/views/Feedstocks.vue'),
      props: true,
      meta: {
        authNotRequired: true,
        backRoute: 'Feedstocks'
      }
    },
    {
      path: '/feedstocks/:categoryId/:id',
      name: 'Feedstock',
      component: () =>
        import(/* webpackChunkName: "client-chunk-feedstock-details" */ '@/views/FeedstockDetails.vue'),
      props: true,
      meta: {
        authNotRequired: true,
        backRoute: 'Material'
      }
    },
    {
      path: '/myfeedstocks',
      name: 'My Feedstocks',
      beforeEnter: (to, from, next) => {
        if (from.name === 'Feedstocks') {
          to.meta.backRoute = 'Feedstocks'
        }
        next()
      },
      props: true,
      component: () =>
        import(/* webpackChunkName: "client-chunk-myFeedstocks" */ '@/views/MyFeedstocks.vue')
    },
    {
      path: '/myfeedstocks/create',
      name: 'Create Feedstock',
      component: () =>
        import(/* webpackChunkName: "client-chunk-myFeedstocks-create" */ '@/views/CreateFeedstock.vue'),
      meta: {
        backRoute: 'My Feedstocks'
      }
    },
    {
      path: '/myfeedstocks/:id',
      name: 'My feedstock',
      component: () =>
        import(/* webpackChunkName: "client-chunk-my-feedstock-details" */ '@/views/FeedstockDetails.vue'),
      props: true,
      meta: {
        backRoute: 'My Feedstocks'
      }
    },
    {
      path: '/calculator',
      name: 'Recipe calculator',
      component: () =>
        import(/* webpackChunkName: "client-chunk-calculator" */ '@/views/Calculator.vue'),
      props: true,
      meta: {
        authNotRequired: true
      }
    },
    {
      path: '/calculator/save-recipe',
      name: 'Save Recipe',
      component: () =>
        import(/* webpackChunkName: "client-chunk-calculator" */ '@/views/SaveRecipe.vue'),
      meta: {
        backRoute: 'Recipe calculator'
      }
    },
    {
      path: '/calculator/myrecipes',
      name: 'My Recipes',
      component: () =>
        import(/* webpackChunkName: "client-chunk-my-recipes" */ '@/views/MyRecipes.vue')
    },
    {
      path: '/settings',
      name: 'Settings',
      component: () =>
        import(/* webpackChunkName: "client-chunk-settings" */ '@/views/Settings'),
      meta: {
        authNotRequired: true
      }
    },
    {
      path: '/about',
      name: 'About',
      component: () =>
        import(/* webpackChunkName: "client-chunk-settings" */ '@/views/About'),
      meta: {
        authNotRequired: true
      }
    },
    {
      path: '*',
      component: () =>
        import(/* webpackChunkName: "client-chunk-404" */ '@/views/PageNotFound')
    }
  ]
})

/**
 * Handle user redirections
 */
// eslint-disable-next-line consistent-return
router.beforeEach((to, from, next) => {
  if (
    !(to.meta && to.meta.authNotRequired) &&
    isNil(store.state.authentication.user)
  ) {
    const path =
      store.state.authentication.user === null ? '/login' : '/check-login'
    return next(`${path}?redirectUrl=${to.path}`)
  }
  next()
})

export default router
