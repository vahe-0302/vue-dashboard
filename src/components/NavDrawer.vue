<template>
  <v-navigation-drawer
    v-model="showDrawerComputed"
    app
    @input="checkHideDrawer"
  >
    <v-list-item>
      <v-list-item-content>
        <v-list-item-title class="title"> Recipe Calculator </v-list-item-title>
        <v-list-item-subtitle>
          {{ user ? user.email : '' }}
        </v-list-item-subtitle>
      </v-list-item-content>
    </v-list-item>

    <v-divider></v-divider>

    <v-list>
      <v-list-item-group v-model="selectedItem">
        <v-list-item
          v-for="item in items"
          :key="item.title"
          link
          @click="item.action"
        >
          <v-list-item-icon>
            <v-icon>{{ item.icon }}</v-icon>
          </v-list-item-icon>

          <v-list-item-content>
            <v-list-item-title>{{ item.title }}</v-list-item-title>
          </v-list-item-content>
        </v-list-item>
      </v-list-item-group>

      <v-list-item v-if="user" link @click="logout">

        <v-list-item-icon>
          <v-icon>mdi-logout</v-icon>
        </v-list-item-icon>

        <v-list-item-content>
          <v-list-item-title>logout</v-list-item-title>
        </v-list-item-content>
      </v-list-item>
    </v-list>
  </v-navigation-drawer>
</template>

<script>
import firebase from 'firebase/app'
import { mapMutations, mapState } from 'vuex'
import router from '../router'

export default {
  props: {
    showDrawer: {
      type: Boolean,
      default: true
    }
  },
  data() {
    return {
      selectedItem: 0,
      items: [
        {
          icon: 'mdi-home',
          title: 'Home',
          action() {
            router.push({ name: 'Home' }).catch(err => {
              console.log('TCL: action -> err', err)
            })
          }
        },
        {
          icon: 'mdi-calculator',
          title: 'Calculator',
          action() {
            router.push({ name: 'Recipe calculator' }).catch(err => {
              console.log('TCL: action -> err', err)
            })
          }
        },
        {
          icon: 'mdi-format-list-bulleted',
          title: 'Feedstocks',
          action() {
            router.push({ name: 'Feedstocks' }).catch(err => {
              console.log('TCL: action -> err', err)
            })
          }
        },
        {
          icon: 'mdi-star',
          title: 'myFeedstocks',
          action() {
            router.push({ name: 'My Feedstocks' }).catch(err => {
              console.log('TCL: action -> err', err)
            })
          }
        },
        {
          icon: 'mdi-chart-bar',
          title: 'myRecipes',
          action() {
            router.push({ name: 'My Recipes' }).catch(err => {
              console.log('TCL: action -> err', err)
            })
          }
        },
        {
          icon: 'mdi-cog',

          title: 'Settings',
          action() {
            router.push({ name: 'Settings' }).catch(err => {
              console.log('TCL: action -> err', err)
            })
          }
        }
      ]
    }
  },
  computed: {
    showDrawerComputed: {
      get() {
        return this.showDrawer
      },
      set() {
        return null
      }
    },
    ...mapState('authentication', ['user'])
  },
  methods: {
    ...mapMutations('snackbar', ['setSnackbar']),
    async logout() {
      await firebase
        .auth()
        .signOut()
        .then(() => {
          this.setSnackbar({
            show: true,
            text: 'You have been signed out!'
          })
        })
    },
    checkHideDrawer(show) {
      if (!show) {
        this.$emit('closeDrawer')
      }
    }
  }
}
</script>

<style lang="scss" scoped></style>
