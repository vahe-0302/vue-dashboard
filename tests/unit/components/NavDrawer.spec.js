// import Vuex from 'vuex'
import { cloneDeep } from 'lodash'
import { createLocalVue, shallowMount } from '@vue/test-utils'
import NavDrawer from '@/components/NavDrawer'
import router from '@/router'
import Vuetify from 'vuetify'

const localVue = createLocalVue()
localVue.use(Vuetify)

describe('NavDrawer', () => {
  test('NavDrawer snapshot test', () => {
    const wrapper = shallowMount(NavDrawer, { localVue, router })
    expect(wrapper).toMatchSnapshot()
  })
})