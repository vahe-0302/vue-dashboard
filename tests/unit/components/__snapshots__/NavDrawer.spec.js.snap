// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavDrawer NavDrawer snapshot test 1`] = `
<v-navigation-drawer-stub app="true" height="100vh" minivariantwidth="80" mobilebreakpoint="1264" src="" tag="nav" width="256" value="true">
  <v-list-item-stub activeclass="" tag="div">
    <v-list-item-content-stub>
      <v-list-item-title-stub class="title">
        Compost calculator
      </v-list-item-title-stub>
      <v-list-item-subtitle-stub>
        subtext
      </v-list-item-subtitle-stub>
    </v-list-item-content-stub>
  </v-list-item-stub>
  <v-divider-stub></v-divider-stub>
  <v-list-stub tag="div" tile="true">
    <v-list-item-group-stub value="0" activeclass="v-item--active">
      <v-list-item-stub activeclass="" link="true" tag="div">
        <v-list-item-icon-stub>
          <v-icon-stub>mdi-home</v-icon-stub>
        </v-list-item-icon-stub>
        <v-list-item-content-stub>
          <v-list-item-title-stub>Home</v-list-item-title-stub>
        </v-list-item-content-stub>
      </v-list-item-stub>
      <v-list-item-stub activeclass="" link="true" tag="div">
        <v-list-item-icon-stub>
          <v-icon-stub>mdi-calculator</v-icon-stub>
        </v-list-item-icon-stub>
        <v-list-item-content-stub>
          <v-list-item-title-stub>Calculator</v-list-item-title-stub>
        </v-list-item-content-stub>
      </v-list-item-stub>
      <v-list-item-stub activeclass="" link="true" tag="div">
        <v-list-item-icon-stub>
          <v-icon-stub>mdi-sprout</v-icon-stub>
        </v-list-item-icon-stub>
        <v-list-item-content-stub>
          <v-list-item-title-stub>Feedstocks</v-list-item-title-stub>
        </v-list-item-content-stub>
      </v-list-item-stub>
      <v-list-item-stub activeclass="" link="true" tag="div">
        <v-list-item-icon-stub>
          <v-icon-stub>mdi-leaf</v-icon-stub>
        </v-list-item-icon-stub>
        <v-list-item-content-stub>
          <v-list-item-title-stub>myFeedstocks</v-list-item-title-stub>
        </v-list-item-content-stub>
      </v-list-item-stub>
      <v-list-item-stub activeclass="" link="true" tag="div">
        <v-list-item-icon-stub>
          <v-icon-stub>mdi-chart-bar</v-icon-stub>
        </v-list-item-icon-stub>
        <v-list-item-content-stub>
          <v-list-item-title-stub>myResults</v-list-item-title-stub>
        </v-list-item-content-stub>
      </v-list-item-stub>
      <v-list-item-stub activeclass="" link="true" tag="div">
        <v-list-item-icon-stub>
          <v-icon-stub>mdi-settings</v-icon-stub>
        </v-list-item-icon-stub>
        <v-list-item-content-stub>
          <v-list-item-title-stub>Settings</v-list-item-title-stub>
        </v-list-item-content-stub>
      </v-list-item-stub>
    </v-list-item-group-stub>
  </v-list-stub>
</v-navigation-drawer-stub>
`;
